frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/common/SideEffectingActionTrackingService.java

```diff
@@ -0,0 +1,32 @@
+package com.virtuslab.gitmachete.frontend.actions.common;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import com.intellij.openapi.components.Service;
+import com.intellij.openapi.project.Project;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.checkerframework.checker.tainting.qual.Untainted;
+
+@Service
+public final class SideEffectingActionTrackingService {
+
+  private final AtomicReference<Set<>> lock = new AtomicReference<>(null);
```

Comment:
_stream of consciousness_

maybe it should be a (multi)Set instead? and no check on action start? just accumulate task ids… so that there’s no problem with task not starting coz its parent took a lock?
but on the other hand, this might cause a successful run when the task should already be disabled… but that’s the problem now as well anyway!